### PR TITLE
Remove SauceCTL and WDIO GH Workflows

### PR DESCRIPTION
**Pull request by spider-sauce**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by lysannep — APPROVED** [review]

File: .github/workflows/deploy-to-production.yml [hunk 1]
```diff
@@ -54,48 +50,6 @@ jobs:
...
-          curl -L -s https://github.com/saucelabs/saucectl/releases/download/v${SAUCECTL_VERSION}/saucectl_${SAUCECTL_VERSION}_linux_64-bit.tar.gz | sudo tar -xvz -C /usr/bin/
-      - name: saucectl run
-        run: |
-          # Expose website
-          cd build/ && python3 -m http.server 8000 &
-
-          saucectl run -c ./.sauce/config.yml --tunnel-name proxy-tunnel-$GITHUB_RUN_ID
-
   deploy-staging:
     needs: test
```
Comment (lysannep) [inline .github/workflows/deploy-to-production.yml:54]: @spider-sauce Should this be updated as well?

Comment (lysannep) [inline .github/workflows/deploy-to-production.yml:54]:
```suggestion
    needs: build
```

**Comment by spider-sauce** [comment]
> @lysannep I also changed the deployment workflow so that it only runs when there are changes to the main branch. This should also reduce the amount of checks required to merge PRs.

Nevermind, apparently the build check is a required status 🤷

**Comment by lysannep** [comment]
@spider-sauce Thank you! :rocket:

**Pull request merged**